lib/modules/manager/npm/extract/common/catalogs.ts

```diff
@@ -0,0 +1,37 @@
+import { PackageDependency } from '../../../types';
+import { NpmManagerData } from '../../types';
+import { Catalog } from '../types';
+import { extractDependency, parseDepName } from './dependency';
+
+export const PNPM_CATALOG_DEPENDENCY = 'pnpm.catalog';
+export const YARN_CATALOG_DEPENDENCY = 'yarn.catalog';
+
+/**
+ * In order to facilitate matching on specific catalogs, we structure the
+ * depType as `[pnpm|yarn].catalog.default`, `[pnpm|yarn].catalog.react17`, and so on.
+ */
+function getCatalogDepType(name: string, npmManager: 'pnpm' | 'yarn'): string {
+  return `${npmManager === 'pnpm' ? PNPM_CATALOG_DEPENDENCY : YARN_CATALOG_DEPENDENCY}.${name}`;
```

Comment:
this probably needs a specific test for full coverage

Reply:
ok will add some